### PR TITLE
Avoid near infinite loops optimizing the typeit font size

### PR DESCRIPTION
**Pull request by cmcgee-mac**
Please note that I've chosen arbitrarily small and large limits to avoid the near infinite loops. I hope that these are suitable. If not, I can adjust to whatever is recommended.

### CODE REVIEW

File: jquery.signaturepad.js [hunk 1]
```diff
@@ -528,14 +528,14 @@ function SignaturePad (selector, options) {
     }
 
     if (typeItNumChars > oldLength && typed.outerWidth() > element.width) {
-      while (typed.outerWidth() > element.width) {
+      while (typeItCurrentFontSize > 4 && typed.outerWidth() > element.width) {
```
Comment (mockdeep) [inline jquery.signaturepad.js:531]: Would this work if we make it `> 0` instead? I'm not sure what the best option is, but it seems like we should allow it so long as it's a positive number.

File: jquery.signaturepad.js [hunk 2]
```diff
@@ -528,14 +528,14 @@ function SignaturePad (selector, options) {
...
         typeItCurrentFontSize--
         typed.css('font-size', typeItCurrentFontSize + 'px')
       }
     }
 
     if (typeItNumChars < oldLength && typed.outerWidth() + edgeOffset < element.width && typeItCurrentFontSize < typeItDefaultFontSize) {
-      while (typed.outerWidth() + edgeOffset < element.width && typeItCurrentFontSize < typeItDefaultFontSize) {
+      while (typeItCurrentFontSize < 512 && typed.outerWidth() + edgeOffset < element.width && typeItCurrentFontSize < typeItDefaultFontSize) {
```
Comment (mockdeep) [inline jquery.signaturepad.js:538]: I only vaguely understand the scenario this PR is trying to fix, but the impression I got was that it was a problem for shrinking the font size. Is it also an issue for large font sizes as well?

**Review by mockdeep — COMMENTED** [review]
@cmcgee-mac thanks for the PR! A couple of questions, but looks pretty straightforward!

File: jquery.signaturepad.js (hunk 1 above)
Comment (cmcgee-mac) [reply jquery.signaturepad.js:531]: I was thinking that it should be at least visible. I can change this to zero if that would be best.

File: jquery.signaturepad.js (hunk 2 above)
Comment (cmcgee-mac) [reply jquery.signaturepad.js:538]: The original problem is that changing the font size has no impact at all on the typed.outerWidth() causing the long looping. I thought that there should be some kind of guard here too. Maybe 512 is not a good number here or perhaps there is a better solution?

Comment (mockdeep) [reply jquery.signaturepad.js:538]: Okay, sounds good. I think this is probably fine for now.

**Comment by mockdeep** [comment]
Thanks for contributing!

**Pull request merged**